internal/mq/msgstream/mqwrapper/kafka/kafka_consumer.go

```diff
@@ -226,9 +221,27 @@ func (kc *Consumer) GetLatestMsgID() (mqwrapper.MessageID, error) {
 	return &kafkaID{messageID: high}, nil
 }
 
+func (kc *Consumer) closeInternal() {
+	log.Info("close consumer ", zap.String("topic", kc.topic), zap.String("groupID", kc.groupID))
+	start := time.Now()
+	err := kc.c.Close()
+	if err != nil {
+		log.Warn("failed to close ", zap.String("topic", kc.topic), zap.Error(err))
+	}
+	cost := time.Since(start).Milliseconds()
+	if cost > 200 {
+		log.Warn("close consumer costs too long time", zap.Any("topic", kc.topic), zap.String("groupID", kc.groupID), zap.Int64("time(ms)", cost))
+	}
+}
+
 func (kc *Consumer) Close() {
 	kc.closeOnce.Do(func() {
-		close(kc.closeCh)
-		kc.wg.Wait() // wait work goroutine exit, wait worker close the client
+		if kc.hasChan {
```

Comment:
What is the purpose of requiring a "hasChan"? How about:
```Go
kc.closeOnce.Do(func() {
    close(kc.closeCh)
    kc.closeInternal()
    kc.wg.Wait()
}
```

For non-blocking close:
```Go
kc.closeOnce.Do(func() {
    close(kc.closeCh)
    go kc.closeInternal()
}
```